### ISSUE TITLE
JS Exception when opening the lobby menu

### ISSUE THREAD

**Issue by BossCode45** [momentum-mod/game#1957]
### Describe the bug

When I click on the arrow at the side to open the lobby the game freezes and I get a JS Exception
![image](https://user-images.githubusercontent.com/74084650/208834303-6e1f2f78-11d9-44cc-ab92-1afe1fe17146.png)

### How To Reproduce

Open game
Click arrow on the side to open lobby menu

### Issue Map

Main menu

### Expected Behavior

There shouldn't be an error

### Operating System

Linux arch 5.15.80-1-lts x86_64

### Renderer

DX11 (default)

**Comment by tsa96** [comment]
Hmm, can't replicate. `Lobby.recreateLobbyListChildren` gets called when the list opens, and expects some data in `Lobby.lobbyListData` - though I don't get this error if I launch the game without an internet connection. Presumably `lobbyTypes` in undefined (though this will break for anything besides `0`, `1`, and `2`), which is derived from the data passed in through C++, and seems that data just wasn't passed in yet, so we should null/undefined check it more.

**Comment by BossCode45** [comment]
Yeah I haven't gotten the same issue in a while (although I'm not sure why), should I mark as closed?

**Comment by tsa96** [comment]
Let's leave it for a bit, plus we should probably do the null checks anyway.

**Comment by tsa96** [comment]
Reared it's head again: https://discord.com/channels/235111289435717633/1129513440919158925
Just need to handle cases where `newType` or whatever is undefined

**Comment by Gocnak** [comment]
Closing this for now, I think we did a good enough job nullchecking this. If it pops up again we can re-open.
